src/components/organisms/PortalAddEditForm/PortalAddEditForm.tsx

```diff
@@ -0,0 +1,268 @@
+  onDone: () => void;
+  portal?: Room;
+  venueVisibility?: RoomVisibility;
+  portalIndex?: number;
+}
+
+export const PortalAddEditForm: React.FC<PortalAddEditFormProps> = ({
+  portal,
+  item,
+  onDone,
+  venueVisibility,
+  portalIndex,
+}) => {
+  const { user } = useUser();
+  const { worldSlug, spaceSlug } = useParams<AdminVenueViewRouteParams>();
+  const { spaceId: currentSpaceId, world } = useWorldAndSpaceBySlug(
+    worldSlug,
+    spaceSlug
+  );
+
+  const { icon } = item ?? {};
+
+  const isEditMode = isTruthy(portal);
+  const title = isEditMode ? "Edit the portal" : "Create a portal";
+
+  const defaultValues = useMemo(
+    () => ({
+      title: portal?.title ?? "",
+      image_url: portal?.image_url ?? icon ?? DEFAULT_VENUE_LOGO,
+      visibility: portal?.visibility ?? venueVisibility,
+      spaceId: portal?.spaceId ?? undefined,
+      isClickable: portal?.type !== RoomType.unclickable,
+      isEnabled: portal?.isEnabled ?? true,
+    }),
+    [
+      portal?.title,
+      portal?.image_url,
+      portal?.visibility,
+      portal?.spaceId,
+      portal?.type,
+      portal?.isEnabled,
+      icon,
+      venueVisibility,
+    ]
+  );
+
+  const {
+    register,
+    getValues,
+    handleSubmit,
+    errors,
+    setValue,
+    reset,
+  } = useForm({
+    reValidateMode: "onChange",
+
+    validationSchema: roomSchema,
+    defaultValues,
+  });
+
+  useEffect(() => reset(defaultValues), [defaultValues, reset]);
+
+  const changeRoomImageUrl = useCallback(
+    (val: string) => {
+      setValue("image_url", val, false);
+    },
+    [setValue]
+  );
+
+  const [
+    { loading: isLoading, error: submitError },
+    addPortal,
+  ] = useAsyncFn(async () => {
+    if (!user || !world || !currentSpaceId) return;
+
+    const {
+      title,
+      image_url,
+      visibility,
+      spaceId,
+      isClickable,
+      isEnabled,
+    } = getValues();
+
+    const portalSource = portal ?? {};
+
+    const portalData: PortalInput = {
+      ...DEFAULT_PORTAL_INPUT,
+      ...portalSource,
+      title,
+      image_url,
+      visibility,
+      spaceId,
+      type: !isClickable ? RoomType.unclickable : undefined,
+      isEnabled,
+    };
+
+    if (portal) {
+      await upsertRoom(portalData, currentSpaceId, user, portalIndex);
+    } else {
+      await createRoom(portalData, currentSpaceId, user);
+    }
+
+    await onDone();
+  }, [currentSpaceId, getValues, onDone, portal, portalIndex, user, world]);
+
+  const { ownedVenues } = useOwnedVenues({});
+
+  const backButtonOptionList = useMemo(
+    () =>
+      Object.fromEntries(
+        ownedVenues
+          .filter(
+            ({ worldId: venueWorldId, id }) =>
+              !(venueWorldId !== world?.id || id === currentSpaceId)
+          )
+          .map((venue) => [venue.id, venue])
+      ),
+    [currentSpaceId, ownedVenues, world?.id]
+  );
+
+  // const [isOverrideAppearanceEnabled, setOverridenAppearance] = useState(false);
```

Comment:
commented the lines with the Override appearance toggle until we resolve the issue which I mentioned in the description to the PR (or if we decide to remove that toggle at all 😬 )